### PR TITLE
feat: manager review questions

### PR DESCRIPTION
**Pull request by mdeljavan**

### CODE REVIEW

File: src/pages/manager-review-page/achievements/ManagerReviewAchievementsRatingGroup.tsx [hunk 1]
```diff
@@ -50,19 +49,18 @@ export const ManagerReviewAchievementsRatingGroup = React.memo(function ManagerR
...
 
-  const peerAnswers = (answers: Answers) =>
-    innerJoin((a, b) => a.questionId === b.id, answers, peerReviewProjectQuestions);
+  const peerAnswers = (answers: Answers) => getQuestionsAnswersPair(peerReviewProjectQuestions, answers);
 
   const filteredByRating = rating
     ? comments.filter((comment) => comment.rating === rating)
-    : comments.filter((comment) => comment.rating === rating && peerAnswers(comment.answers).every(prop('value')));
-  const filteredComments = filteredByRating.filter((comment) => peerAnswers(comment.answers).every(prop('value')));
+    : comments.filter((comment) => comment.rating === rating && peerAnswers(comment.answers).hasPairs);
```
Comment (amirhossein-asadi) [inline src/pages/manager-review-page/achievements/ManagerReviewAchievementsRatingGroup.tsx:56]: comment.rating === rating 
to
!comment.rating

**Review by amirhossein-asadi — APPROVED** [review]

File: src/pages/manager-review-page/achievements/ManagerReviewAchievementsExpansionPanel.tsx [hunk 2]
```diff
@@ -86,19 +92,19 @@ export function ManagerReviewAchievementsExpansionPanel(props: Props) {
...
+              <ManagerProjectOutput self review={projectReview.managerComment} />
+            </Grid>
+          )}
+          {!printing && (
+            <Grid item xs={12}>
+              <FormChangeDetector>
+                <ManagerReviewForm onSubmit={saveManagerProjectReview} formData={projectReview.managerComment} />
+              </FormChangeDetector>
+            </Grid>
+          )}
```
Comment (amirhossein-asadi) [inline src/pages/manager-review-page/achievements/ManagerReviewAchievementsExpansionPanel.tsx:107]: i think use ternary operator here is better

File: src/pages/manager-review-page/dominant-characteristics/ManagerReviewDominantCharacteristicsExpansionPanel.tsx [hunk 3]
```diff
@@ -27,6 +28,10 @@ interface OwnProps {
   reviews: ManagerReviewDominantCharacteristicsExpansionPanel_reviews$key;
   title?: string;
   type: 'strengths' | 'weaknesses';
+  managerPersonReview: {
+    readonly strengths: ReadonlyArray<string> | null;
```
Comment (amirhossein-asadi) [inline src/pages/manager-review-page/dominant-characteristics/ManagerReviewDominantCharacteristicsExpansionPanel.tsx:32]: the first readonly isn't enough?

File: src/shared/project-output/ManagerProjectOutput.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  const me = useAuthGuardUser();
+  const questionsAnswers = getQuestionsAnswersPair(managerReviewProjectQuestions, review.answers);
+
+  if (!review) {
+    return null;
+  }
+
+  return (
+    <Box bgcolor="success.light" borderRadius={3} p={2}>
+      <ReviewItemInfo name={getUserLabel(me)} src={me.avatarUrl ?? undefined} type="manager">
```
Comment (amirhossein-asadi) [inline src/shared/project-output/ManagerProjectOutput.tsx:47]: in ManagerReviewDominantCharacteristicsExpansionPanel.tsx u use name={getUserLabel(me) 
?? undefined} . i think u can delete this there
maybe u can delete this for me.avatarUrl ?? undefined too

File: src/shared/review-item-info/AnonymousReviewItemInfo.tsx [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { Box, Grid, lighten, useTheme } from '@material-ui/core';
 import { FCProps } from 'src/shared/types/FCProps';
 
 interface OwnProps {
-  type: 'self' | 'peer';
+  type: 'self' | 'peer' | 'manager';
```
Comment (amirhossein-asadi) [inline src/shared/review-item-info/AnonymousReviewItemInfo.tsx:8]: i think u can declare a type for this and use it

File: src/shared/review-item-info/IdentifiedReviewItemInfo.tsx [hunk 6]
```diff
@@ -18,34 +18,39 @@ import { i18n } from '@lingui/core';
 interface OwnProps {
   name?: string;
   src?: string;
-  type: 'self' | 'peer';
+  type: 'self' | 'peer' | 'manager';
```
Comment (amirhossein-asadi) [inline src/shared/review-item-info/IdentifiedReviewItemInfo.tsx:21]: also here

File: src/shared/review-item-info/IdentifiedReviewItemInfo.tsx [hunk 7]
```diff
@@ -18,34 +18,39 @@ import { i18n } from '@lingui/core';
...
+      <Grid container spacing={2} className={classes.root}>
+        <Grid item>
+          <Avatar className={classes.avatar} src={src}>
+            {name?.[0]}
+          </Avatar>
+        </Grid>
+        <Grid item xs>
+          <Box marginBottom={1}>
+            <Typography variant="button">
+              {name} <Typography variant="caption">{type === 'manager' && `(${i18n._('Manager')})`}</Typography>
```
Comment (amirhossein-asadi) [inline src/shared/review-item-info/IdentifiedReviewItemInfo.tsx:42]: why this typography is child?
they have different variant

File: src/shared/review-item-info/ReviewItemInfo.tsx [hunk 8]
```diff
@@ -8,7 +8,7 @@ interface OwnProps {
   anonymous?: boolean;
   name?: string;
   src?: string;
-  type: 'self' | 'peer';
+  type: 'self' | 'peer' | 'manager';
```
Comment (amirhossein-asadi) [inline src/shared/review-item-info/ReviewItemInfo.tsx:11]: here too

File: src/pages/manager-review-page/dominant-characteristics/ManagerReviewDominantCharacteristicsExpansionPanel.tsx (hunk 3 above)
Comment (mdeljavan) [reply src/pages/manager-review-page/dominant-characteristics/ManagerReviewDominantCharacteristicsExpansionPanel.tsx:32]: the first `readonly` is for strengths, you can't change its value i.e `managerPersonReview.strengths = []` and second `readonly` for strengths value that you can't mutate its value.
[for more info](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgLIBM7IN4ChkHJQRzoD2IANgJ7IDOYxIA5mABZ0BcyASieVWoBBKFDjUAPAyihmAPlwBfXLgQUGyALbcMWALw58haRBbsuyANoByAIzWANNYBM1gLpKVmgHQmzHZANLD1wfP1YObwAHAFc6NgAKawAWawBKIA)

File: src/shared/project-output/ManagerProjectOutput.tsx (hunk 4 above)
Comment (mdeljavan) [reply src/shared/project-output/ManagerProjectOutput.tsx:47]: I want to use this component in the result phase, so I've designed it more generically.

File: src/shared/review-item-info/IdentifiedReviewItemInfo.tsx (hunk 7 above)
Comment (mdeljavan) [reply src/shared/review-item-info/IdentifiedReviewItemInfo.tsx:42]: Goog point

File: src/pages/manager-review-page/achievements/ManagerReviewAchievementsExpansionPanel.tsx (hunk 2 above)
Comment (mdeljavan) [reply src/pages/manager-review-page/achievements/ManagerReviewAchievementsExpansionPanel.tsx:107]: if I use the ternary operator, we have a two-step indentation. IMO this is more readable.

File: src/shared/review-item-info/IdentifiedReviewItemInfo.tsx [hunk 9]
```diff
@@ -15,37 +15,43 @@ import { FCProps } from 'src/shared/types/FCProps';
...
+    <Box bgcolor={bgColor} clone>
+      <Grid container spacing={2} className={classes.root}>
+        <Grid item>
+          <Avatar className={classes.avatar} src={src}>
+            {name?.[0]}
+          </Avatar>
+        </Grid>
+        <Grid item xs>
+          <Box marginBottom={1}>
+            <Typography variant="button">{name}</Typography>{' '}
```
Comment (amirhossein-asadi) [inline src/shared/review-item-info/IdentifiedReviewItemInfo.tsx:43]: remove {' '}

Comment (mdeljavan) [reply src/shared/review-item-info/IdentifiedReviewItemInfo.tsx:43]: I need this

**Review by amirhossein-asadi — APPROVED** [review]

**Pull request merged**